charms/istio-pilot/src/charm.py

```diff
@@ -124,12 +124,26 @@ def send_info(self, event):
 
     def handle_ingress(self, event):
         try:
-            self._get_gateway_address
+            if self._get_gateway_address:
+                self.unit.status = ActiveStatus()
```

Comment:
Can we comfortably set an active status here if this statement is true?

Reply:
I think that makes sense.  And possibly at the top of this charm there could also be a `MaintenanceStatus`, but I haven't thought that through much so I'll go with your gut on whether it is needed.  

fwiw, this charm still has an `ActiveStatus` in the `__init__` so all these nuances will get trampled whenever an unattached hook fires, but that's a future problem for us :)

Reply:
I suggest we think about statuses better and discuss them in a follow up PR.